test_conformance/c11_atomics/common.h

```diff
@@ -487,16 +491,11 @@ class CBasicTestMemOrderScope : public CBasicTest<HostAtomicType, HostDataType>
     std::vector<TExplicitMemoryScopeType> memoryScope;
     int error = 0;
 
-    memoryOrder.push_back(MEMORY_ORDER_EMPTY);
-    memoryOrder.push_back(MEMORY_ORDER_RELAXED);
-    memoryOrder.push_back(MEMORY_ORDER_ACQUIRE);
-    memoryOrder.push_back(MEMORY_ORDER_RELEASE);
-    memoryOrder.push_back(MEMORY_ORDER_ACQ_REL);
-    memoryOrder.push_back(MEMORY_ORDER_SEQ_CST);
-    memoryScope.push_back(MEMORY_SCOPE_EMPTY);
-    memoryScope.push_back(MEMORY_SCOPE_WORK_GROUP);
-    memoryScope.push_back(MEMORY_SCOPE_DEVICE);
-    memoryScope.push_back(MEMORY_SCOPE_ALL_SVM_DEVICES);
+    // For OpenCL-3.0 and later some orderings and scopes are optional, so here
+    // we query for the supported ones.
+    test_error_ret(
+        getSupportedMemoryOrdersAndScopes(deviceID, memoryOrder, memoryScope),
```

Comment:
I reviewed another PR that used a similar pattern earlier today.  In short, I think this will work, but as-written the test_error_ret macro will call the function twice if the first call failed, which might not be expected.

If we're going to use this pattern more regularly then I think we should modify the macro so it assigns to a temporary error code so this pattern will only call the function once.

This doesn't need to be done as part of this PR.

Reply:
Thanks for the review! Ah yes, good catch, it's no problem to update the `test_error_ret` macro as part of this MR, I have pushed a patch assigning the parameter to a temporary value so if it is a function call it will only happen once.